feature-odyssey/src/main/kotlin/com/hedvig/android/odyssey/search/groups/GetNetworkClaimEntryPointGroupsUseCase.kt

```diff
@@ -0,0 +1,35 @@
+package com.hedvig.android.odyssey.search.groups
+
+import arrow.core.Either
+import arrow.core.raise.either
+import com.apollographql.apollo3.ApolloClient
+import com.hedvig.android.apollo.safeExecute
+import com.hedvig.android.apollo.toEither
+import com.hedvig.android.core.common.ErrorMessage
+import octopus.EntrypointGroupsQuery
+
+internal class GetNetworkClaimEntryPointGroupsUseCase(
+    private val apolloClient: ApolloClient,
+) {
+
+  suspend fun invoke(): Either<ErrorMessage, GroupedClaimsResult> {
```

Comment:
Return type could even be `Either<ErrorMessage, List<ClaimGroup>` directly and could remove GroupedClaimsResult if we don't use GroupedClaimsResult in any other significant way, but no hard opinions here.